haskell/private/cabal_wrapper.py.tpl

```diff
@@ -143,6 +143,18 @@ with tmpdir() as distdir:
         "--with-hc-pkg=" + ghc_pkg,
         "--with-ar=" + ar,
         "--with-gcc=" + cc,
+        "--ghc-option=-pgmc=" + cc,
```

Comment:
Could this re-use [`ghc_cc_program_args`](https://github.com/cocreature/rules_haskell/blob/a3d4a5759c31380d810a0916e248cb9a6a6fda49/haskell/cc.bzl#L164) to avoid these flags diverging in future?
Using something like `"%{ghc_cc_args}": repr(ghc_cc_program_args(cc))` [here](https://github.com/cocreature/rules_haskell/blob/a3d4a5759c31380d810a0916e248cb9a6a6fda49/haskell/cabal_wrapper.bzl#L31).

Reply:
Ah that looks good, I’ll give it a shot. Wasn’t quite sure how to best pass this down.

Reply:
I’ve tried it but just passing it on doesn’t quite work. I’m missing the `find_exe` logic from the python script in that case so `cc` points to the wrong location. I’m not really sure what to do about this. I could imagine some logic in the python script that applies `find_exe` to flag that don’t start with `-` but that seems worse than inlining the flags here. Any ideas?

Reply:
Indeed, that's unfortunate. `ghc_cc_program_args` just expects a string representing cc, so we could also use a place holder and fill it in on the python side. Something like this:
`bzl`:
```
    "%{ghc_cc_args}": repr(ghc_cc_program_args("$CC"))
```
`python`:
```
    ] +
    [ arg.replace("$CC", cc) for arg in %{ghc_cc_args} ] +
    [
```

That's getting on the verge, but seems still better than duplicating the flags.

Reply:
Clever, I like it! I’ve made the change.